content/chapter-destinations/destination-s3/_index.md

```diff
@@ -0,0 +1,205 @@
+| Type:    | string |
+| Default: | `5120GiB` |
+
+*Description:* The maximal size of the S3 object.
+<!-- What happens if we reach it, we start a new object? -->
+
+## max-pending-uploads()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer |
+| Default: | `32` |
+
+*Description:* <!-- FIXME -->
+
+## object-key()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The [object key](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html) (or key name), which uniquely identifies the object in an Amazon S3 bucket.
+
+## object-key-timestamp()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The timestamp added to the object.
+<!-- FIXME ?? -->
+
+## region()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The [regional endpoint](https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints) where the bucket is stored. For example, `us-east-1`
+
+## secret-key()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: |  |
+
+*Description:* The `SECRET_KEY` of the service account used to access the S3 bucket. (Together with [`access-key()`](#access-key).)
+
+## storage-class()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | string |
+| Default: | `STANDARD` |
+
+*Description:* The [storage class of the object](https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-class-intro.html), for example, `REDUCED_REDUNDANCY`. The following values are valid:
+
+`DEEP_ARCHIVE`, `GLACIER`, `GLACIER_IR`, `INTELLIGENT_TIERING`, `ONEZONE_IA`, `OUTPOSTS`, `REDUCED_REDUNDANCY`, `SNOW`, `STANDARD`, `STANDARD_IA`
+
+If you configure an invalid value, the default is used.
+
+<!-- FIXME do we have any recommendation on which one is good for storing logs? -->
+
+## upload-threads()
+
+|          |                            |
+| -------- | -------------------------- |
+| Type:    | integer |
+| Default: | `8` |
+
+*Description:* The number of {{% param "product_name" %}} worker threads that are used to upload data to S3 from this destination.
+
+## template()
+
+|          |                               |
+| -------- | ----------------------------- |
+| Type:    | template or template function |
+| Default: | `${MESSAGE}\n` |
+
+*Description:* The message as written to the Amazon S3 object. You can use templates and [template functions]({{< relref "/chapter-manipulating-messages/customizing-message-format/reference-template-functions/_index.md" >}}) to format the message.
+
+<!-- FIXME What else do we write into the object? Timestamp/and so on? -->
```

Comment:
Only the resolution of the `template()` is written to the file. However the `object-key()` can be templated to factor out common fields, like host, cluster, container, etc... Also `object-key-timestamp()` can be used to add timestamp related macros.